components/brave_wallet_ui/common/async/lib.ts

```diff
@@ -1085,11 +1086,8 @@ export function refreshPortfolioFilterOptions () {
       )
     }
 
-    if (
-      selectedAccountFilter !== AllAccountsOption.id &&
-      !accounts.some(account => account.id === selectedAccountFilter)
-    ) {
-      dispatch(WalletActions.setSelectedAccountFilterItem(AllAccountsOption.id))
+    if (!applySelectedAccountFilter(accounts, selectedAccountFilter).accounts) {
+      dispatch(WalletActions.setSelectedAccountFilterItem(AllAccountsOptionUniqueKey))
```

Comment:
Moved similar account filtering algorithms to a single place. Maybe that should be a hook in the future.